src/lightning_lite/accelerators/cuda.py

```diff
@@ -78,10 +78,63 @@ def register_accelerators(cls, accelerator_registry: Dict) -> None:
         )
 
 
-def _get_all_available_cuda_gpus() -> List[int]:
+def find_usable_cuda_devices(num_devices: int = -1) -> List[int]:
+    """Returns a list of all available and usable CUDA GPUs.
```

Comment:
```suggestion
    """Returns a list of all available and usable CUDA devices.
```

Reply:
I will add some of these, but from the context it should be very clear that GPU means NVIDIA CUDA. 
GPU is not a forbidden word. We can use it and people are very familiar with it. There is nothing wrong with the current description.

> A CUDA device is a GPU but a GPU might not be a CUDA device (e.g. MPS)

You can take this further, into the absurd:

A CUDA device is a GPU but a GPU might not be a CUDA device (e.g. MPS), unless it is an AMD device then it is again pretending to be a CUDA GPU in PyTorch unless you are on MacOS with an AMD GPU then it is actually the MPS device in PyTorch.